### PR TITLE
fix(enrichment): keep quarantined descriptions out of tailoring while staying scoreable and visible

### PR DESCRIPTION
**Pull request by ebarti**
> Stacked on #214; review/merge after #214. Base branch is #214's head (`worktree-agent-a289889308712f9f9`), so the diff shown here is scoped to this PR.

## Problem (verified)
The enrich path (`enrichment/detail.py`) computes a `SnapshotConfidence` and writes LOW-confidence rows to `discovery_quarantine_entries`, but `succeed_attempt` marks the `JobEnrichment` aggregate `enriched` regardless. Scoring/materials read `je.full_description` via a COALESCE gated **only** on active state (`database.py` `_NOT_CLOSED_ACTIVE_STATE`) — no selector checked confidence/quarantine. So LOW-confidence / quarantined descriptions fed scoring **and** tailoring/apply as if trustworthy, defeating the quarantine and violating the invariant that a job declared "usable" meets a real quality bar.

## Fix (recall-safe — never silently drop a good job)
1. **Surface** the enrichment quality signal downstream (auditability):
   - Persist the latest snapshot's `confidence` + `quarantine_reason` on `posting_snapshot_sets` (`latest_confidence` / `latest_quarantine_reason`) via an idempotent migration + one-time JSON backfill; `SqlitePostingSnapshotSetRepository.save` writes them going forward.
   - Promote `enrichment_confidence` / `enrichment_quarantine_reason` onto the job read model (`get_jobs_by_stage`, `load_job_with_enrichment`).
   - Add `confidence` / `quarantine_reason` / `quarantined` to the `PostingContentSnapshotCaptured` event payload, and render Confidence + Quarantine (warning tone) on the job-detail audit entry (`apps/api/src/read-model.ts`).
2. **Gate only the expensive, employer-facing steps** on the explicit LOW-confidence quarantine marker: tailoring, cover-letter, and apply prep selectors (`get_jobs_by_stage` `pending_tailor`/`pending_cover`/`pending_apply`, the pipeline runner `_count_pending`/`_PENDING_SQL`, `get_stats` `untailored_eligible`/`ready_to_apply`, `count_ready_to_apply`, and the apply launcher batch selection). Scoring stays **ungated** — quarantined jobs remain scoreable and visible.
3. **Do NOT gate on a merely missing apply URL** (a different, recoverable condition, per the adversarial-reviewer warning): a MEDIUM/HIGH snapshot keeps `latest_confidence` above LOW and still tailors; an operator-overridden LOW snapshot (reason `none`) also passes. The gate is `latest_confidence = 'low' AND latest_quarantine_reason NOT IN ('none','')`.

## Quarantine marker gated on
The latest `PostingContentSnapshot`'s persisted `latest_confidence`/`latest_quarantine_reason` — authoritative and self-healing (a confident re-enrich clears the gate automatically), unlike the `discovery_quarantine_entries` review queue which is not cleared on re-enrich.

## Tests
New `workers/automation/tests/test_enrichment_quarantine_gating.py`:
- LOW-confidence quarantined enriched job is EXCLUDED from `pending_tailor` but PRESENT in `scored`, and visible via `load_job_with_enrichment` + list rows with the quality signal.
- LOW-confidence enriched job is still selected for `pending_score` (scoreable).
- Normal-confidence job flows through `scored` AND `pending_tailor` unchanged.
- MEDIUM/HIGH snapshot missing its apply URL is NOT gated; operator-overridden LOW snapshot is NOT gated.
- `untailored_eligible` stat excludes the quarantined job; repository persists the columns; migration backfills legacy rows; the `PostingContentSnapshotCaptured` event carries confidence/quarantine.
- New `apps/api/test/server.test.ts` case asserts the audit entry surfaces Confidence + Quarantine with warning tone.

## Verification
- `uv --project workers/automation run --extra dev pytest -q workers/automation/tests/test_enrichment_snapshot_pr3.py workers/automation/tests/test_materials_queue_selectors.py workers/automation/tests/test_score_queue_selectors.py workers/automation/tests/test_preparation_work_items.py workers/automation/tests/test_enrichment_use_cases.py workers/automation/tests/test_enrichment_quarantine_gating.py` → **79 passed**.
- Full Python suite: **1503 passed**.
- `ruff check` on changed files → **All checks passed!**
- `pnpm --filter @jobhunter/api check` → clean; `pnpm --filter @jobhunter/api test` → **286 passed**.

## Docs
- `docs/architecture.md` — new "enrichment quality gate" paragraph.
- `docs/local-reliability-qa.md` — regression-matrix entry.

### CODE REVIEW

File: workers/automation/src/jobhunter/database.py [hunk 1]
```diff
@@ -2608,6 +2658,22 @@ def _backfill_one_observation_row(
...
+# extraction is not trustworthy enough to spend the expensive, employer-facing
+# tailoring / cover / apply steps on. It stays scoreable (cheap triage) and
+# visible with its quality signal, so a quarantined job never vanishes from the
+# funnel. Only a genuinely LOW-confidence quarantine is gated: a MEDIUM/HIGH
+# snapshot missing its apply URL is quarantined for review but keeps
+# ``latest_confidence`` above LOW, so a recoverable missing field never starves
+# tailoring. An operator-overridden LOW snapshot carries reason 'none' and also
+# passes. Reads through the ``pss`` alias exposed by ``_ACTIVE_STATE_JOIN``.
+_ENRICHMENT_NOT_QUARANTINED: str = (
+    "(pss.latest_confidence IS NULL "
```
Comment (ebarti) [inline workers/automation/src/jobhunter/database.py:2774]: PR Reviewer - pr-reviewer

Severity: Medium
Gate: Pass
Evidence: This predicate's `pss.latest_confidence IS NULL` disjunct is the load-bearing recall guarantee: with the LEFT JOIN in `_ACTIVE_STATE_JOIN`, every job that has NO posting_snapshot_sets row (the entire pre-feature enriched backlog, plus any legacy row the backfill left with NULL confidence) gets pss.latest_confidence = NULL and must pass the gate. But every test in test_enrichment_quarantine_gating.py seeds a snapshot row, so no test exercises the NULL branch. A future NULL-unsafe "simplification" (e.g. collapsing this to `pss.latest_confidence != 'low'` — in SQL `NULL != 'low'` is NULL/false) would silently starve the ENTIRE no-snapshot backlog out of tailoring/cover/apply, and the suite would stay green. Given recall is the #1 priority, this branch deserves a pinned regression test.
Reproduction: Verified the current behavior is correct — ran an enriched+scored job with no posting_snapshot_sets row: it appears in get_jobs_by_stage(stage="pending_tailor") (not starved). Also verified self-heal: a LOW+quarantine snapshot is gated, then re-recording a HIGH/NONE snapshot clears the gate (latest snapshot authoritative). Both pass today; neither is covered by a test.

Required fix: Add a regression test for the no-snapshot / NULL-confidence case — an enriched + scored job with NO posting_snapshot_sets row must be PRESENT in pending_tailor (and pending_cover/apply) — so the NULL-safety of this predicate can never silently regress. A self-heal test (LOW snapshot gated -> re-enrich HIGH -> ungated) would also pin the "latest snapshot authoritative" behavior. Test-only; the production logic is already correct.

Comment (ebarti) [reply workers/automation/src/jobhunter/database.py:2774]: Done in d48b4db (test-only). Added two hardening tests to `test_enrichment_quarantine_gating.py` — the file now has **11 tests** (was 9):

1. **`test_no_snapshot_backlog_job_is_never_gated_from_any_selector`** — an enriched + scored job with **no** `posting_snapshot_sets` row (`latest_confidence IS NULL` — the entire pre-feature backlog) must be PRESENT in `pending_tailor`, `pending_cover`, `pending_apply`, and counted in `untailored_eligible` / `ready_to_apply` / `count_ready_to_apply`.
   **Fail-if-broken confirmed:** I temporarily flipped `_ENRICHMENT_NOT_QUARANTINED` to the NULL-unsafe `(pss.latest_confidence != 'low')` and this test failed — the backlog job was starved out of the tailoring queue (`AssertionError: assert 'https://example.com/backlog/untailored' in set()`). Reverted; the predicate diff is now empty (byte-identical to the reviewed form).

2. **`test_reenrichment_to_high_self_heals_a_quarantined_job_into_tailoring`** — a job quarantined LOW (gated out) then re-enriched to a HIGH / non-quarantined snapshot returns to `pending_tailor`, and the read model reflects the healed confidence. Pins the "latest snapshot authoritative" contract (fails if the gate ever switched to an "any snapshot ever quarantined" check).

Verification: `pytest -q workers/automation/tests/test_enrichment_quarantine_gating.py` → **11 passed**; `ruff check` on the changed files → clean.
